app/fileManager.js

```diff
@@ -51,6 +52,15 @@ function exists(id) {
     xhr.send();
   });
 }
+*/
+
+function getDLCounts(file) {
+  return new Promise((resolve, reject) => {
+    const url = `/api/metadata/${file.id}`;
+    const receiver = new FileReceiver(url, file);
+    resolve(receiver.getMetadata(file.nonce));
+  });
+}
```

Comment:
Aside from other changes you might want to make I'd like to see the style use `async/await` instead of `Promise`. i.e.

```js
async function getDLCounts(file) {
  const url = `/api/metadata/${file.id}`;
  const receiver = new FileReceiver(url, file);
  try {
    const result = await receiver.getMetadata(file.nonce)
  }
  // ...
  return // something?
}
```

Reply:
This again is not working. When i Try to logthe results, they are blank

Reply:
> try {
    const result = await receiver.getMetadata(file.nonce)
  }

Apparently, `getMetaData` doesn't return anything and so the `result` is always null. Are you talking about `fetchMetaData`?

Reply:
I'm not saying to use that exact code, it's just an example. I want you to use async/await.

Reply:
oh